[UXUI-239] Implement email preview on details page

andersonjeccel opened the pull request:
| Q                                      | A
| -------------------------------------- | --------------------------
| Bug fix?          | ❌
| New feature/enhancement?      | ✔️
| Deprecations?                          | ❌
| BC breaks?        | ❌
| Automated tests included?              | ❌ 
| Related user documentation PR URL      | mautic/user-documentation#...
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #...

## Description

Adds the email content preview and header details to the email details page.

Context: I received user feedback saying that a quick preview directly on the details screen would be positive because in a scenario where you're comparing 5 emails to locate what you're looking for, having to open the builder or adding extra tabs to check the entire email preview was an inefficient workflow

<img width="1214" height="730" alt="image" src="https://github.com/user-attachments/assets/0e84b07c-d104-438a-a953-cd6a57ed8f63" />




---
### 📋 Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester)).
2. Create or open an existing email, then open its details view.
3. Confirm the new preview header layout shows subject, sender information, and labels (To, Reply to, Bcc depending on what you have configured in your email).
4. Confirm the email body preview iframe loads correctly within the details page.

---
<!--
If you have any deprecations and backwards compatibility breaks, list them here along with the new alternative.
-->

andersonjeccel committed:
Add preview block

andersonjeccel committed:
Remove old details

andersonjeccel committed:
MIigrate to trans strings

andersonjeccel committed:
Spacing improvements

andersonjeccel committed:
code quality

andersonjeccel commented:
@kuzmany My vision is that other tabs on the screen are related to data (stats / map), so Preview could look strange among them (and potentially lose its usefulness if not the first tab)...

(added a bit of context to the PR description)

kuzmany commented:
@andersonjeccel I meant this set of tabs; it could be the first in the row. This set contains heatmaps in our app and make sense for me.


<img width="1316" height="388" alt="image" src="https://github.com/user-attachments/assets/3ae7a755-f2b5-4ccf-aec5-e7a3135ac439" />

escopecz committed:
Merge branch '7.x' into enhancement/Implement-email-preview-on-details-page

escopecz commented:
@andersonjeccel this is pretty cool. But this feature to replace tokens should also work, right?
<img width="1483" height="870" alt="Screenshot 2025-12-12 at 16 02 28" src="https://github.com/user-attachments/assets/9df42717-d49d-478c-b27d-e524252b3016" />

rabiibrahimi commented:
> Shouldn't this go to the tab?

I agree - i think it will be  better to be placed in a tab. Having this one main page of the email could be disturbing as its purpose is to only preview and you only preview emails before sent but once emails is sent you are more focused on stats etc.

hluchas commented on app/bundles/EmailBundle/Resources/views/Email/details.html.twig:
```diff
@@ -225,21 +225,15 @@
-                  <div class="box-layout">
-                      <div class="col-xs-10">
-                          <div>{{ email.subject }}</div>
-                          <div class="text-secondary">{{ emailPreview }}</div>
```
This is left in the controller as dead code.

andersonjeccel replied:
done!

hluchas commented on app/bundles/EmailBundle/Resources/views/Email/Details/preview.html.twig:
```diff
@@ -0,0 +1,85 @@
+        <span class="type-label-01">&lt;{{ email.fromAddress|default(configGetParameter('mailer_from_email')) }}&gt;</span>
+      </div>
+      <span class="type-label-01 text-secondary">{{ 'mautic.email.preview.to_label'|trans }}</span>
+      <span class="type-label-02">{{ 'mautic.email.preview.to_me'|trans }}</span>
```
Suggestion (not needed for approval)

The email is not sent "To: Me" — it is sent to segment contacts. On the details page of an already-sent email this reads as incorrect information. Remove the To: row entirely, or replace it with the segment name(s) or recipient count.

andersonjeccel replied:
done!

hluchas reviewed (changes requested):
Please remove dead code from controller.
Work well.
Maybe token replacement in preview will be handy.
TY!

andersonjeccel committed:
Merge branch '7.x' into enhancement/Implement-email-preview-on-details-page

andersonjeccel committed:
simplify ui

andersonjeccel committed:
remove dead code from email details controller

andersonjeccel committed:
replace deprecated scrolling attribute with css overflow

andersonjeccel committed:
Merge branch '7.x' into enhancement/Implement-email-preview-on-details-page

andersonjeccel committed:
Merge branch '7.x' into enhancement/Implement-email-preview-on-details-page

andersonjeccel committed:
Merge branch '7.x' into enhancement/Implement-email-preview-on-details-page

hluchas reviewed (changes requested):
Looks good now.

One thing: 
1. Create segment email.
2. Do not enter the builder and save it.
3. Preview is not visible.
4. Go to the builder and do nothing. Then close it.
5. Save the email.
6. Preview is visible.

andersonjeccel committed:
fix email preview fallback

andersonjeccel committed:
Merge branch '7.x' into enhancement/Implement-email-preview-on-details-page

hluchas reviewed (approved):
Seems to be working well.
TY!

andersonjeccel committed:
Merge branch '7.x' into enhancement/Implement-email-preview-on-details-page

andersonjeccel committed:
Merge branch '7.x' into enhancement/Implement-email-preview-on-details-page

andersonjeccel committed:
Merge branch '7.x' into enhancement/Implement-email-preview-on-details-page

escopecz committed:
Merge branch '7.x' into enhancement/Implement-email-preview-on-details-page

andersonjeccel committed:
adjust preview scaling

andersonjeccel committed:
resolve conflicts

andersonjeccel committed:
Merge branch '7.x' into enhancement/Implement-email-preview-on-details-page

kuzmany committed:
[UXUI-239] Sandbox email preview iframe to prevent stored XSS

The new details-page preview iframe loaded the same-origin preview URL
without a sandbox, so stored email HTML containing inline <script>
(which survives InputHelper::html sanitization) executed in the admin's
authenticated session when the details page was opened. Add
sandbox="allow-popups allow-popups-to-escape-sandbox" (omitting
allow-scripts and allow-same-origin) to block script execution and
same-origin access while keeping the visual HTML/CSS preview intact.

Signed-off-by: Zdeno Kuzmany <zdeno@kuzmany.biz>

kuzmany reviewed (approved):
Everything looks good to me 👍
